Sources/SourceKitLSP/Swift/CodeActions/PackageManifestEdits.swift

```diff
@@ -0,0 +1,254 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import PackageModel
+import PackageModelSyntax
+import SwiftRefactor
+import SwiftSyntax
+
+/// Syntactic code action provider to provide refactoring actions that
+/// edit a package manifest.
+struct PackageManifestEdits: SyntaxCodeActionProvider {
+  static func codeActions(in scope: SyntaxCodeActionScope) -> [CodeAction] {
+    guard let token = scope.firstToken,
+      let call = token.findEnclosingCall()
+    else {
+      return []
+    }
+
+    var actions = [CodeAction]()
+
+    // Add test target(s)
+    actions.append(
+      contentsOf: maybeAddTestTargetActions(call: call, in: scope)
+    )
+
+    // Add product(s).
+    actions.append(
+      contentsOf: maybeAddProductActions(call: call, in: scope)
+    )
+
+    return actions
+  }
+
+  /// Produce code actions to add test target(s) if we are currently on
+  /// a target for which we know how to create a test.
+  static func maybeAddTestTargetActions(
+    call: FunctionCallExprSyntax,
+    in scope: SyntaxCodeActionScope
+  ) -> [CodeAction] {
```

Comment:
It’s not really “adding” actions to anything right? What about just adding `addTestTargetTestActions`?

Reply:
Eh, I guess so.